applications/HDF5Application/python_scripts/core/controllers.py

```diff
@@ -15,17 +15,18 @@
 
 # STL imports
 import abc
+import typing
 
 
 ##!@addtogroup HDF5Application
 ##!@{
 ##!@name Kratos classes
 ##!@{
-class Controller(metaclass=abc.ABCMeta):
+class Controller(abc.ABC):
     def __init__(self,
                  model_part: KratosMultiphysics.ModelPart,
                  operation: operations.AggregateOperation):
-        self.model_part = model_part
+        self._model_part = model_part
         self.__operation = operation
```

Comment:
`model_part` and `operation` are function arguments that already have type hints, so the member variables should already have the correct type info. Is it not working in your IDE?

Reply:
it might be detected by the IDE, but I am pretty sure that type-checkers like `mypy` would complain

Reply:
are you using e.g. mypy locally? I would suggest to do that (until it is integrated in the CI).

Reply:
> are you using e.g. mypy locally? I would suggest to do that (until it is integrated in the CI).

Yup I have `mypy` locally, so stubs are automatically generated every time I compile, and they work just fine. My IDE relies on the generated stubs, and it does pick up the type of `self._model_part` even without the explicit hint.

Reply:
I mean do you do `mypy applications/HDF5Application/python_scripts/core/controllers.py`?

Reply:
Nope, I imagine every dependency (including Kratos) would have to be correctly hinted for that to work. Am I wrong about that?

Reply:
yes :)
although you can selectively disable some checks, so that it can be added later also in established frameworks. 

My point is, I think randomly adding the type annotations without enforcing them will lead to a tough time when doing them properly. Aside from them being potentially wrong...

I recommend to give mypy a try, to see what errors can pop up etc. My writing of python code has changed quite a bit since I started to use it

Reply:
Pun intended? 😛